rfcs/text/0428-extending-types.md

```diff
@@ -0,0 +1,677 @@
+        "$ref": "https://blockprotocol.org/@alice/property-type/birth-date/v/1"
+      }
+    }
+  }
+}
+```
+
+given that the original schema was created as follows
+
+```json
+{
+  "$id": "https://blockprotocol.org/@alice/entity-type/person/v/1",
+  "type": "object",
+  "kind": "entityType",
+  "title": "Person",
+  "properties": {
+    "https://blockprotocol.org/@alice/property-type/name": {
+      "$ref": "https://blockprotocol.org/@alice/property-type/name/v/1"
+    },
+    "https://blockprotocol.org/@alice/property-type/email": {
+      "$ref": "https://blockprotocol.org/@alice/property-type/email/v/1"
+    },
+    "https://blockprotocol.org/@alice/property-type/phone-number": {
+      "$ref": "https://blockprotocol.org/@alice/property-type/phone-number/v/1"
+    }
+  }
+}
+```
+
+The above change applies to these existing BP operations:
+
+- `updateEntityType`
+- `updatePropertyType`
+- `updateLinkType`
+
+which must all make use of complete schemas in place of partial ones. These examples originate from the [Type System RFC](./0352-graph-type-system.md#interfacing-with-types-1).
+
+While not explained in detail in the Type System RFC the `aggregateEntities` operation would need to change behavior slightly. The types returned from aggregating on a specific entity type could now also be implicitly coerced instances of entities (supertypes projected from subtype instances).
+
+# Drawbacks
+
+[drawbacks]: #drawbacks
+
+- The way this proposal adds type extension means that we must implement some version of property selection/projection for types, which comes with non-trivial implementation details for embedding applications.
+- This drifts further away from JSON Schema by introducing a different meaning to the `unevaluatedProperties` keyword and using it implicitly for our schemas.
+- The way compatibility is defined could result in newer versions of supertypes becoming incompatible over time. This is not unique to extending types and can be an "issue" in many parts of the type system as different versions of property types become incompatible.
+
+# Rationale and alternatives
+
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+The general rationale for this way of handling extended types (which we may also call type inheritance) is that we want to keep supertypes and subtypes compatible with one another. Constraining the way type inheritance works makes it so that we can implicitly have "same as" relations across type inheritance trees. It also allows extending multiple supertypes which can lead to more expressive domain models.
+
+## Problems and alternatives
+
+`unevaluatedProperties` _almost_ provides the functionality we're after, but unfortunately it just barely misses. If supertypes themselves specify `{ "unevaluatedProperties": false }`, they are not able to be part of an `allOf` validator, as they will error out as soon as they see properties that are not part of the supertype itself.
```

Comment:
I feel like this explanation would be useful higher up where we talk about changes to `unevaluatedProperties`- the explanation for what we're doing with it and why feels a bit scattered at the moment.

We could have this explanation higher up, but keep the 'alternative' proposal down here.

Reply:
It was originally in the explanation section, but moved it down here. I can move it back up

Reply:
Moved in 82f9032c84db30d8c6c19b80c2c8a784f9636163